enhancements/installer/ipi-publishing-status.md

```diff
@@ -0,0 +1,206 @@
+parameter as it was set in the install-config.yaml.
+
+## Proposal
+
+The Infrastructure API is extended, adding the `Publish` field to the
+`InfrastructureStatus`.
+
+```go
+type InfrastructureStatus struct {
+	// ...
+
+	// publish indicates the publishing status of the cluster at install time.
+	// The default is 'External', which means the cluster will be publicly
+    // available on the internet.
+	// The 'Internal' mode indicates a private cluster, and operators should
+    // configure operands accordingly.
+    // +kubebuilder:default=External
+    // +kubebuilder:validation:Enum=Internal;External
+	Publish PublishMode `json:"publish"`
+}
+```
+
+To reflect the install-config publishing options, `PublishMode` will only
+support to valid options.
+
+```
+// PublishMode defines the publish mode of the cluster at install time.
+type PublishMode string
+
+const (
+	// "Internal" is for operators to configure their asserts as private
+    // when appropriate.
+	InternalPublishMode PublishMode = "Internal"
+
+	// "External" indicates that the cluster is publicly available on the internet.
+	ExternalPublishMode PublishMode = "External"
+)
+```
+
+### Workflow Description
+
+1. The cluster creator install a cluster via IPI, setting `publish: Internal` in installer-config.yaml
+2. The installer bootstraps the cluster, and sets `Publish: Internal` in the Infrastructure status
+3. Operators use this value to configure operands accordingly.
+
+### API Extensions
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+The `Publish` field in the infrastructure status will only reflect the initial
+status of the cluster. Individual components (like ingress) may be changed on
+day-2 to for example turn a public cluster private, and the `Publish` field in
+the Infrastructure status will remain untouched.
+
+#### Hypershift [optional]
+
+Hypershift publishing options are not an exact match for the ones provided by
+the installer in installer-config.yaml.
+
+Hypershift options (limited to AWS) are:
+
+* `Public`
+* `PublicAndPrivate`
+* `Private`
+
+`PublicAndPrivate` does not translate well to OpenShift. It allows public API
+server access and private node communication with the control plane, making
+the cluster appear public to end-users, while securing control plane and
+nodes communication.
+
+The suggested approach to Hypershift is to translate the modes as follows:
+
+* `Public` -> `Public`
+* `PublicAndPrivate` -> `Public`
+* `Private` -> `Private`
+
+### Risks and Mitigations
+
+Users may interpret the `Publish` field in the Infrastructure status as a way
+to change the publishing status of their clusters at run time, which is not
+its intention.
```

Comment:
Users should not be updating the status, if you want to make the field immutable, that can be done with CEL or admission policy (the latter if you want to enable a particular user (eg a controller) to update the field, but no one else)

Reply:
Can you give me a reference to CEL? first time I hear about it.
This field is not intended to be updated by anyone after it's written, so we don't need an admission policy at this point.

Reply:
Sure, take a look here, https://kubernetes.io/docs/reference/using-api/cel/

And as an example of an immutable field, https://github.com/openshift/api/blob/f404f2bc3524e03f3bc29a7b3cd0f624cd36eae7/example/v1/types_stable.go#L48-L58

Reply:
thanks! applied to code sample.